src/main/java/com/spring_boot/services/PropertiesService.java

```diff
@@ -0,0 +1,69 @@
+package com.spring_boot.services;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Service;
+
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+@Service
+public class PropertiesService {
```

Comment:
you're trying to reinvent the weel here. the logic of reading and merging properties from different sources is managed  by spring boot. All you need to care about is either `Environment` bean or `@Value` annotation to access properties.